tests/unit/lib/build_module/test_app_builder.py

```diff
@@ -739,6 +739,35 @@ def test_can_build_image_function_under_debug(self, mock_os):
             ),
         )
 
+    @patch("samcli.lib.build.app_builder.os")
+    def test_can_build_image_function_under_debug_with_target(self, mock_os):
+        mock_os.environ.get.return_value = "debug"
+        metadata = {
+            "Dockerfile": "Dockerfile",
+            "DockerContext": "context",
+            "DockerTag": "Tag",
+            "DockerBuildArgs": {"a": "b"},
+            "DockerBuildTarget": "stage",
+        }
+
+        self.docker_client_mock.api.build.return_value = []
+
+        result = self.builder._build_lambda_image("Name", metadata)
+
+        self.assertEqual(result, "name:Tag-debug")
+        self.builder._build_lambda_image("Name", metadata)
```

Comment:
Is the duplicate call necessary?

Reply:
I dont think its necessary.

Reply:
could we address this?

Reply:
It also appears in another test case: https://github.com/aws/aws-sam-cli/blob/49851f58d5b64f65d48d127ef9a5fc4eccf96ed9/tests/unit/lib/build_module/test_app_builder.py#L729